overlord/snapstate/handlers.go

```diff
@@ -3528,6 +3587,74 @@ func (m *SnapManager) doRemoveAliases(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+func (m *SnapManager) undoRemoveAliases(t *state.Task, _ *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+	snapsup, snapst, err := snapSetupAndState(t)
+	if err != nil {
+		return err
+	}
+
+	if !snapst.AliasesPending {
+		// do nothing
+		return nil
+	}
+
+	snapName := snapsup.InstanceName()
+	curAliases := snapst.Aliases
+	_, _, err = applyAliasesChange(snapName, autoDis, nil, snapst.AutoAliasesDisabled, curAliases, m.backend, doApply)
+	if err != nil {
+		return err
+	}
+
+	snapst.AliasesPending = false
+	Set(st, snapName, snapst)
+	return nil
+}
+
+func shouldPruneOldAliases(t *state.Task, snapst *SnapState) (prune bool, oldAliases map[string]*AliasTarget, oldAutoDisabled bool, err error) {
+	chg := t.Change()
+	for _, t := range chg.Tasks() {
```

Comment:
given that we do not expect to change the fact that set-auto-aliases is just before setup-aliases, this could use t.WaitTasks() instead of going over the entire Change thasks

Reply:
that's a nice optimization, thank you!